packages/opencensus-core/src/resource/resource.ts

```diff
@@ -0,0 +1,192 @@
+  private readonly labels: StringMap;
+
+  private static readonly MAX_LENGTH = 255;
+  private static readonly LABEL_LIST_SPLITTER = ',';
+  private static readonly LABEL_KEY_VALUE_SPLITTER = '=';
+  private static readonly ENV_TYPE =
+      Resource.parseResourceType(process.env.OC_RESOURCE_TYPE);
+  private static readonly ENV_LABEL_MAP =
+      Resource.parseResourceLabels(process.env.OC_RESOURCE_LABELS);
+  private static readonly ERROR_MESSAGE_INVALID_CHARS =
+      'should be a ASCII string with a length greater than 0 and not exceed ' +
+      Resource.MAX_LENGTH + ' characters.';
+  private static readonly ERROR_MESSAGE_INVALID_VALUE =
+      'should be a ASCII string with a length not exceed ' +
+      Resource.MAX_LENGTH + ' characters.';
+
+  constructor(type: string, labels: StringMap) {
+    this.type = type;
+    this.labels = labels;
+  }
+
+  /**
+   * Returns a Resource. This resource information is loaded from the
+   * OC_RESOURCE_TYPE and OC_RESOURCE_LABELS environment variables.
+   *
+   * @returns {Resource}
+   */
+  static createFromEnvironmentVariables(): Resource {
+    return new Resource(Resource.ENV_TYPE, Resource.ENV_LABEL_MAP);
+  }
+
+  /**
+   * Returns a Resource that runs all input resources sequentially and merges
+   * their results. In case a type of label key is already set, the first set
+   * value takes precedence.
+   *
+   * @param  {Resource[]} resources the list of the resources.
+   * @returns {Resource}
+   */
+  static mergeResources(resources: Resource[]): Resource {
+    let currentResource: Resource = null;
+    for (const resource of resources) {
+      currentResource = this.merge(currentResource, resource);
+    }
+    return currentResource;
+  }
+
+  /**
+   * Returns the type identifier for the resource.
+   *
+   * @returns {string}
+   */
+  getType(): string {
+    return this.type;
+  }
+
+  /**
+   * Returns a map of labels that describe the resource.
+   *
+   * @returns {StringMap}
+   */
+  getLabels(): StringMap {
+    return this.labels;
+  }
+
+  /**
+   * Creates a resource type from the OC_RESOURCE_TYPE environment variable.
+   *
+   * <p>OC_RESOURCE_TYPE: A string that describes the type of the resource
+   * prefixed by a domain namespace, e.g. “kubernetes.io/container”.
+   */
+  private static parseResourceType(rawEnvType: string): string {
+    if (rawEnvType) {
+      if (!Resource.isValidAndNotEmpty(rawEnvType)) {
+        throw new Error(`Type ${Resource.ERROR_MESSAGE_INVALID_CHARS}`);
+      }
+      return rawEnvType.trim();
+    }
+    return null;
+  }
+
+  /**
+   * Creates a label map from the OC_RESOURCE_LABELS environment variable.
+   *
+   * <p>OC_RESOURCE_LABELS: A comma-separated list of labels describing the
+   * source in more detail, e.g. “key1=val1,key2=val2”. Domain names and paths
+   * are accepted as label keys. Values may be quoted or unquoted in general. If
+   * a value contains whitespaces, =, or " characters, it must always be quoted.
```

Comment:
Please add doc for params and return

Reply:
In general, I don't think we add params and return for private methods.

Reply:
It will help for other developers

Reply:
Ok, done.